src/components/organisms/AddForm/AddForm.jsx

```diff
@@ -0,0 +1,71 @@
+import { Formik, Form, Field } from 'formik';
+import * as Yup from 'yup';
+import PropTypes from 'prop-types';
+import { Icon } from '../..';
+
+import './Form.scss';
+
+export const AddForm = ({ placeholder, type }) => {
+    const submitOnEnter = (event) => {
+        if (event.key === 'Enter' && event.shiftKey === false) {
+            event.preventDefault();
+            event.target.form.dispatchEvent(
+                new Event('submit', { cancelable: true, bubbles: true })
+            );
+        }
+    };
+
+    const onSubmit = (values, action) => {
+        // TODO temporary, connect with API
+        action.resetForm({
+            values: {
+                [`${type}`]: ''
+            }
+        });
+    };
+
+    const focusOnInput = (event) => {
+        if (!event.target) {
+            event.target[0].focus();
+        }
+    };
+
+    const validation = Yup.object().shape({
+        [`${type}`]: Yup.string().required('Required')
+    });
+
+    return (
+        <div
+            className={`form form__${type}`}
+            role='directory'
+            onClick={focusOnInput}
+        >
+            <Formik
+                initialValues={{ name: type }}
+                validationSchema={validation}
+                onSubmit={onSubmit}
+            >
+                {({ handleSubmit, errors, touched }) => (
```

Comment:
inline function in jsx is kinda bad, the react would rerender that function each time and create new instance with every click actually, I know that in app such as we are doing here, we for sure won't have any issues with performance, but it should be defined above return statement and wrapped with useCallback

Reply:
Done